app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt

```diff
@@ -207,8 +218,6 @@ class TabTrayView(
     override fun onTabSelected(tab: TabLayout.Tab?) {
         toggleFabText(isPrivateModeSelected)
         filterTabs.invoke(isPrivateModeSelected)
-
-        updateState(view.context.components.core.store.state)
```

Comment:
Why do we no longer need to update state in this case?

Reply:
I didn't see why we needed it until you asked hehe. Added the right UI calls below this :)